packages/ods/src/components/pagination/src/components/ods-pagination/ods-pagination.tsx

```diff
@@ -0,0 +1,331 @@
+import type { EventEmitter, Fragment, FunctionalComponent } from '@stencil/core';
+import { Component, Element, Event, Host, Listen, Method, Prop, State, Watch, forceUpdate, h } from '@stencil/core';
+import { getRandomHTMLId } from '../../../../../utils/dom';
+import { ODS_BUTTON_COLOR, ODS_BUTTON_SIZE, ODS_BUTTON_VARIANT } from '../../../../button/src';
+import { ODS_ICON_NAME } from '../../../../icon/src';
+import { ODS_TEXT_PRESET } from '../../../../text/src';
+import { ODS_PAGINATION_PER_PAGE, ODS_PAGINATION_PER_PAGE_OPTIONS } from '../../constants/pagination-per-page';
+import { computeActualTotalPages, createPageList, getActualPage } from '../../controller/ods-pagination';
+
+@Component({
+  shadow: true,
+  styleUrl: 'ods-pagination.scss',
+  tag: 'ods-pagination',
+})
+export class OdsPagination {
+  private actualTotalPages = 1;
+  private isFirstLoad: boolean = true;
+  private leftArrowButtonId = 'pagination-left-arrow';
+  private rightArrowButtonId = 'pagination-right-arrow';
+  private hostId: string = '';
+  private maxVisibleItems = 7;
+  private ellipsisThreshold = 4;
+
+  @Element() el!: HTMLElement;
+
+  @State() itemPerPage = ODS_PAGINATION_PER_PAGE.option_10;
+  @State() pageList: OdsPaginationPageList = [];
+  @State() current: number = 1;
+
+  @Prop({ reflect: true }) public defaultCurrentPage: number = 1;
+  /** @docType OdsPaginationPerPage */
+  @Prop({ reflect: true }) public defaultItemsPerPage: ODS_PAGINATION_PER_PAGE = ODS_PAGINATION_PER_PAGE.option_10;
+  @Prop({ reflect: true }) public isDisabled: boolean = false;
+  @Prop({ reflect: true }) public labelTooltipNext?: string;
+  @Prop({ reflect: true }) public labelTooltipPrevious?: string;
+  @Prop({ reflect: true }) public totalItems?: number;
+  @Prop({ reflect: true }) public totalPages: number = 1;
+
+  @Event() odsPaginationChanged!: EventEmitter<OdsPaginationChangedEventDetail>;
+  @Event() odsPaginationItemPerPageChanged!: EventEmitter<OdsPaginationItemPerPageChangedEventDetail>;
+
+  @Listen('odsChange')
+  odsValueChangeHandler(event: CustomEvent<OdsSelectEventChangeDetail>): void {
+    event.preventDefault();
+    event.stopPropagation();
+
+    const { value } = event.detail;
+
+    if (value) {
+      this.itemPerPage = typeof value === 'number' ? value : parseInt(value, 10);
+    }
+  }
+
+  @Watch('labelTooltipNext')
+  @Watch('labelTooltipPrevious')
+  updateLabelTooltip(): void {
+    forceUpdate(this.el);
+  }
+
+  @Watch('current')
+  async onCurrentChange(current: number, oldCurrent?: number): Promise<void> {
+    this.updatePageList();
+    this.emitChange(current, oldCurrent);
+  }
+
+  @Watch('totalPages')
+  async onTotalPagesChange(): Promise<void> {
+    if (!this.totalItems) {
+      this.actualTotalPages = this.totalPages;
+    }
+    this.updatePageList();
+  }
+
+  @Watch('itemPerPage')
+  async onItemPerPageChange(): Promise<void> {
+    if (!this.isFirstLoad) {
```

Comment:
Watch are not supposed to be executed on initial load, did you try to remove it to check if it was still needed?